e2e/tests/test_e2e.py

```diff
@@ -694,13 +728,23 @@ def get_service_type(self, svc_labels, namespace='default'):
     def check_service_annotations(self, svc_labels, annotations, namespace='default'):
         svcs = self.api.core_v1.list_namespaced_service(namespace, label_selector=svc_labels, limit=1).items
         for svc in svcs:
-            if len(svc.metadata.annotations) != len(annotations):
+            if len(svc.metadata.annotations) < len(annotations):
                 return False
             for key in svc.metadata.annotations:
                 if svc.metadata.annotations[key] != annotations[key]:
                     return False
         return True
 
+    def check_statefulset_annotations(self, sset_labels, annotations, namespace='default'):
+        ssets = self.api.apps_v1.list_namespaced_stateful_set(namespace, label_selector=sset_labels, limit=1).items
+        for sset in ssets:
```

Comment:
Why loop if only one stateful set is fetched ?
It is effectively `sset = self.api.apps_v1.list_namespaced_stateful_set(namespace, label_selector=sset_labels, limit=1).items[0]`